app/lib/subversion_repository.rb

```diff
@@ -57,15 +57,21 @@ def self.create(connect_string, with_hooks: false)
   # Static method: Opens an existing Subversion repository
   # at location 'connect_string'
   def self.open(connect_string)
+    unless SubversionRepository.repository_exists?(location)
+      raise "Could not open repository at location #{location}"
+    end
     return SubversionRepository.new(connect_string)
   end
 
   # Static method: Yields an existing Subversion repository and closes it afterwards
   def self.access(connect_string)
-    repository = self.open(connect_string)
-    yield repository
-  ensure
-    repository.close
+    raise 'Respository does not exist' unless SubversionRepository.repository_exists? connect_string
+    begin
```

Comment:
You can remove the begin and do `repository&.close` in the ensure.

Reply:
I think we need the begin in this case because we don't want the ensure to trigger if the RuntimeError is raised (before the `repository` variable is initialized)

Reply:
The `&.close` would just do nothing in that case, but again this is only style, both versions are valid. I was just trying to be consistent with Rubocop.

Reply:
Disregard